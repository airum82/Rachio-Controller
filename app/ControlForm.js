import React, { Component } from 'react';

class ControlForm extends Component {
  constructor() {
    super()
    this.state = {
      sortOrder: [],
      zones: [],
      type: '',
      duration: '0',
      zoneId: ''
    }
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    this.setState({
      [e.target.name] : e.target.value
    })
  }

  render() {
    return (
      <div className="form">
        <form onSubmit={(event) => this.props.runAllZones(this.state.duration, event)}>
          <input type="text" name="duration" onChange={(e) => this.handleChange(e)}/>
          <button>run all zones</button>
        </form>
      </div>
    )
  }
}

export default ControlForm;