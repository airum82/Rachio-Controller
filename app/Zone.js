import React, { Component } from 'react';
import { apiKey } from '../APIkey';

class Zone extends Component {
  constructor() {
    super();
    this.state = {
      showForm: false,
      duration: 0,
      selected: false,
      error: ''
    };
    this.toggleForm = this.toggleForm.bind(this);
    this.startZone = this.startZone.bind(this);
    this.handleInput = this.handleInput.bind(this);
    this.handleSelect = this.handleSelect.bind(this);
  }

  toggleForm() {
    this.setState({ showForm: !this.state.showForm });
  }

  handleInput(event) {
    this.setState({
      [event.target.name]: event.target.value
    })
  }

  handleSelect() {
    this.props.selectZone(this.props.id)
    this.setState({ selected: !this.state.selected })
  }

  startZone(event) {
    event.preventDefault();
    const body = {
      id: this.props.id,
      duration: parseInt(this.state.duration)
    };
    return fetch('https://api.rach.io/1/public/zone/start', {
      method: 'PUT',
      headers: {
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(body)
    })
    .then(response => response.text())
    .then(result => console.log(result))
    .then(() => this.toggleForm())
    .catch(error => this.setState({ error: error.message }))
  }

  render() {
    return (
      <div className="zone" id={this.props.id}>
        <h3 className="zone-title">{this.props.name}</h3>
        <img 
          className={this.state.selected ? 'selected' : ''}
          onClick={this.handleSelect}
          src={this.props.image} 
          alt="picture of this particular zone" 
        />
        <div className="zone-buttons">
          <button 
            onClick={this.toggleForm}
            className="toggle-form"
          >
            {this.state.showForm ? 'cancel' : 'start zone'}
          </button>
          <button 
            onClick={this.handleSelect}
            className="select-zone"
          >
            {this.state.selected ? 'unselect' : 'select zone'}
          </button>
        </div>
       <form 
        className={this.state.showForm ? 'zone-form' : 'hidden-form'}
        onSubmit={this.startZone}
       >
        <h4>duration</h4>
        <input name="duration" type="text" onChange={this.handleInput}/>
        <button>start</button>
       </form>
      </div>
    )
  }
}

export default Zone;