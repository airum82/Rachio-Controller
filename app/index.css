@import url('https://fonts.googleapis.com/css?family=Roboto+Condensed');

/*Main page settings*/
body {
  font-family: 'Roboto Condensed', sans-serif;
  color: #2E4057;
}

#app {
  width: 100%;
  height: 100%;
}

.main {
  text-align: center;
  width: 100%;
  height: 100vh;
}

.main::after {
  content: "";
  display: block;
  position: fixed;
  z-index: -1;   
  width: 100%;
  height: 100%;
  background-image: url('../background.jpg');
  background-size: cover;
  opacity: 0.5;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.title {
  margin: 0;
  background-color: #46ACC2;
  position: fixed;
  bottom: 89%;
  width: 100%;
  padding: 2% 0;

}

.devices-intro {
  margin-top: 12%;
}

/*devices styling*/
.device-title {
  font-size: 2em;
}

.devices-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
  align-content: flex-start;
  height: 75vh;
  margin: 0 auto;
  width: 95%;
}

.devices {
  margin: 2%;
  background-color: #42F2F7;
}

.device-link {
  display: block;
  text-decoration: none;
  color: #2E4057;
  font-weight: 700;
  font-size: 1em;
  width: 25vw;
  height: 15vh;
  line-height: 6em;
}

/*zones section styling */
.zones-container {
  display: flex;
  flex-wrap: wrap;
  width: 80%;
  height: 80vh;
  overflow: scroll;
  margin: 0 auto;
  justify-content: space-between;
  font-size: 1em;

}

.zone {
  flex-basis: 32%;
}

.zones-container img {
  display: block;
  height: 40vh;
  margin: 0 auto;
  width: 100%;
}

.zone-title {
  font-size: 1em;
}

.zone-buttons {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.selected {
  border: 4px solid yellow;
}

/*button general styling*/
button:hover {
  cursor: pointer;
}

button {
  border: none;
  background-color: #46ACC2;
}

.back-button {
  display: block;
  margin: 2% auto;
  padding: 1% 3%;
}

/*forms styling */
.hidden-form {
  display: none;
}

.control-form {
  display: flex;
  margin-bottom: 2%;
  justify-content: center;
}

/*media breakpoints*/
@media only screen and (max-width: 790px) {

  .title {
    position: static;
  }

  .devices-intro {
    margin-top: 10%;
    margin-bottom: 5%;
  }

  .devices-container {
    align-content: center;
    flex-direction: column;
    flex-wrap: nowrap;
    height: initial;
  }

  .device-link {
    font-size: 1.2em;
    width: 60vw;
    line-height: 5em;
    margin: 0 auto;
  }

  .zones-container {
    flex-direction: column;
    flex-wrap: nowrap;
  }

  .devices {
    width: inherit;
  }

}

@media only screen and (max-width: 700px) {

  .zone-buttons button {
    padding: 2%;
  }

}

@media only screen and (max-width: 570px) {

  .devices-container {
    align-content: flex-start;
    height: 90vh;
  }

  .form {
    width: 23vh;
    margin: 0 auto;
  }

  .devices-container {
    margin-top: 25%;
  }

}

@media only screen and (max-width: 440px) {

  .devices {
    width: 85%;
  }

}

@media only screen and (max-width: 400px) {

  .device-link {
    font-size: 1em;
  }

}