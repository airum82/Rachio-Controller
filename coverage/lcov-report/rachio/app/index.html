<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for rachio/app</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> rachio/app
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">53.03% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>35/66</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">52.27% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>23/44</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">53.03% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>35/66</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="App.js"><a href="App.js.html">App.js</a></td>
	<td data-value="92.31" class="pic high"><div class="chart"><div class="cover-fill" style="width: 92%;"></div><div class="cover-empty" style="width:8%;"></div></div></td>
	<td data-value="92.31" class="pct high">92.31%</td>
	<td data-value="13" class="abs high">12/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="90.91" class="pct high">90.91%</td>
	<td data-value="11" class="abs high">10/11</td>
	<td data-value="92.31" class="pct high">92.31%</td>
	<td data-value="13" class="abs high">12/13</td>
	</tr>

<tr>
	<td class="file low" data-value="ControlForm.js"><a href="ControlForm.js.html">ControlForm.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	</tr>

<tr>
	<td class="file high" data-value="Device.js"><a href="Device.js.html">Device.js</a></td>
	<td data-value="95" class="pic high"><div class="chart"><div class="cover-fill" style="width: 95%;"></div><div class="cover-empty" style="width:5%;"></div></div></td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="91.67" class="pct high">91.67%</td>
	<td data-value="12" class="abs high">11/12</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	</tr>

<tr>
	<td class="file medium" data-value="DevicesContainer.js"><a href="DevicesContainer.js.html">DevicesContainer.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	</tr>

<tr>
	<td class="file low" data-value="Zone.js"><a href="Zone.js.html">Zone.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	</tr>

<tr>
	<td class="file low" data-value="ZonesContainer.js"><a href="ZonesContainer.js.html">ZonesContainer.js</a></td>
	<td data-value="33.33" class="pic low"><div class="chart"><div class="cover-fill" style="width: 33%;"></div><div class="cover-empty" style="width:67%;"></div></div></td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Oct 03 2018 13:37:35 GMT-0600 (MDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
