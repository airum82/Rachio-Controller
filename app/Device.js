import React, { Component } from 'react';
import ZonesContainer from './ZonesContainer'
import { apiKey } from '../APIkey';
import ControlForm from './ControlForm';

class Device extends Component {
  constructor() {
    super();
    this.state = {
      selectedZones: [],
      zoneIdList: []
    }
    this.runAllZones = this.runAllZones.bind(this);
    this.selectZone = this.selectZone.bind(this);
  }

  runAllZones(duration, event) {
    event.preventDefault()
    const body = {
      zones: this.state.zoneIdList.map((id, index) => {
        return {
          id,
          duration: parseInt(duration),
          sortOrder: index + 1
        }
      })
    }
    console.log(body)
    fetch("https://api.rach.io/1/public/zone/start_multiple", {
      method: 'PUT',
      headers: {
        'Content-Type' : 'application/json',
        'Authorization': `Bearer ${apiKey}`
      },
      body: JSON.stringify(body)
    })
    .then(response => response.text())
    .then(result => console.log(result))
    .catch(error => console.log(error.message))
  }

  selectZone(id) {
    const isSelected = this.state.selectedZones.some(zone => {
      return zone === id
    })
    if(isSelected) {
      this.setState({
        selectedZones: this.state.selectedZones.filter(zone => {
          return zone !== id
        })
      })
    } else {
      this.setState({ 
        selectedZones: [...this.state.selectedZones, id]
      })
    }
  }

  componentDidMount() {
    this.setState({
      zoneIdList: this.props.zones.map(zone => {
        return zone.id
      })
    })
  }

  render() {
    return (
      <div className="device" id={this.props.id}>
        <h2>{this.props.name}</h2>
        <button onClick={this.props.history.goBack}>back</button>
        <ControlForm runAllZones={this.runAllZones}/>
        <ZonesContainer 
          zones={this.props.zones}
          selectZone={this.selectZone}
        />
      </div>
    )
  }
}

export default Device
