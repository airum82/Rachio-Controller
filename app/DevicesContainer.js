import React from 'react';
import Device from './Device';
import { NavLink, Route } from 'react-router-dom';

const DevicesContainer = ({ devices }) => {
  return (
    <div className="devices-container">
      { devices.map((device, index) => {
          return (
            <div className="devices" key={index}>
              <Route exact path='/' render={() => {
                return (
                  <NavLink to={`/${device.id}`} className="device-link">
                    {device.name + ': ' + device.status}
                  </NavLink>
                )
              }} />
              <Route exact path={`/${device.id}`} render={({ history }) => {
                return (
                  <Device
                    id={device.id}
                    name={device.name}
                    zones={device.zones}
                    history={history}
                  />
                )
              }} />
            </div>
          )
        })}
    </div>
  )
}

export default DevicesContainer;