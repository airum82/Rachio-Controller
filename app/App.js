import React, { Component } from 'react';
import { apiKey } from '../APIkey';
import sprinklers from '../sprinklers.gif';
import DevicesContainer from './DevicesContainer';
import { BrowserRouter, Route } from 'react-router-dom';
import './index.css';
import 'normalize.css';

export class App extends Component {
  constructor () {
    super()
    this.state = {
      userInfo: {},
      isLoading: true,
      error: ''
    }
  }

  retrievePersonInfo (id) {
    return fetch(`https://api.rach.io/1/public/person/${id}`, {
      headers: {
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json'
      }
    })
      .then(response => response.json())
      .then(result => this.setState({ userInfo: result, isLoading: false }))
      .catch(error => this.setState({ error: error.message }))
  }

  componentDidMount () {
    fetch('https://api.rach.io/1/public/person/info', {
      headers: {
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json'
      }
    })
      .then(response => response.json())
      .then(result => this.retrievePersonInfo(result.id))
      .catch(error => this.setState({ error: error.message }))
  }

  render () {
    const { devices, fullName } = this.state.userInfo
    return (
      <div className="main">
        {this.state.isLoading ?
          <div className="loading">
            <h1 className="title">Retrieving Devices Now</h1>
            <img src={sprinklers} alt="moving sprinklers to watch while we fetch your content" />
          </div>
          : <div className="main-display">
            <h1 className="title">Hello {fullName}</h1>
            <Route
              exact path="/"
              render={() => {
                return (
                  <h2 className="devices-intro">Your Devices:</h2>
                )
              }}
            />
            <DevicesContainer devices={devices} />
          </div >
        }
      </div>
    )
  }
}