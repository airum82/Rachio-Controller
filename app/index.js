import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { apiKey } from '../APIkey';
import sprinklers from '../sprinklers.gif'
import DevicesContainer from './DevicesContainer';
import './index.css'

class App extends Component {
  constructor() {
    super()
    this.state = {
      userInfo: {},
      isLoading: true
    }
  }

  retrievePersonInfo(id) {
    fetch(`https://api.rach.io/1/public/person/${id}`, {
      headers: {
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json'
      }
    })
    .then(response => response.json())
    .then(result => this.setState({ userInfo: result, isLoading: false }))
    .then(error => console.log(error))
  }

  componentDidMount() {
    fetch('https://api.rach.io/1/public/person/info',{
      headers: { 
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json'
      }
    })
      .then(response => response.json())
      .then(result => this.retrievePersonInfo(result.id))
      .catch(error => console.log(error))
  }

  render() {
    const { devices } = this.state.userInfo
    return this.state.isLoading ? 
    (
      <div>
        <h1>Retrieving Devices Now</h1>
        <img src={sprinklers} alt="moving sprinklers to watch while we fetch your content" />
      </div>
    ) : (
      <div>
        <h1>Hello Arram!</h1 >
        <DevicesContainer devices={devices} />
      </div >
    )
  }
}

ReactDOM.render(
  <App />,
  document.getElementById('app')
);