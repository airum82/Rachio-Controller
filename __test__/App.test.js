import { App } from '../app/App';
import React from 'react';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />, {
      disableLifecycleMethods: true
    })
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve({
        id: 5864
      })
    }));
  })

  it('should match snapshot upon initial render', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot when isLoading is set to false', () => {
    wrapper.instance().setState({ isLoading: false});
    expect(wrapper).toMatchSnapshot();
  })

  it('retrieve person info should call fetch', () => {
    const mockId = 586;
    wrapper.instance().retrievePersonInfo(mockId);
    expect(window.fetch).toHaveBeenCalled();
  })

  it('retrieve person info should set state with userinfo', () => {
    const mockId = 586;
    const spy = jest.spyOn(wrapper.instance(), 'setState');
    return wrapper.instance().retrievePersonInfo(mockId)
      .then(() => {
        expect(spy).toHaveBeenCalledWith({ isLoading: false, userInfo: { id: 5864} });
      })
  })

  it('should set state with error if fetch is unsuccessful', () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.reject({
        message: 'user info not available'
    }))
    const mockId = 586;
    const spy = jest.spyOn(wrapper.instance(), 'setState');
    return wrapper.instance().retrievePersonInfo(mockId)
      .then(() => {
        expect(spy).toHaveBeenCalledWith({ error: 'user info not available' });
      })
  })

  it('componentDidMount should call fetch', () => {
    wrapper = shallow(<App />);
    expect(window.fetch).toHaveBeenCalled();
  })

  it('componentDidMount should call retrievePersonInfo with correct params', async () => {
    wrapper = await shallow(<App />);
    const spy = await jest.spyOn(wrapper.instance(), 'retrievePersonInfo');
    expect(spy).toHaveBeenCalledWith(5864)
  })

  it('componentDidMount should set state with error message if unsuccessful', async () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.reject({
      message: 'user info not available'
    }))
    wrapper = await shallow(<App />);
    const spy = await jest.spyOn(wrapper.instance(), 'setState');
    await wrapper.update();
    expect(spy).toHaveBeenCalledWith({ error: 'user info not available' });
  })

})