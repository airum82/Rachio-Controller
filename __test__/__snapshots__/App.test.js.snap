// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot upon initial render 1`] = `
<div
  className="main"
>
  <div
    className="loading"
  >
    <h1
      className="title"
    >
      Retrieving Devices Now
    </h1>
    <img
      alt="moving sprinklers to watch while we fetch your content"
      src="test-file-stub"
    />
  </div>
</div>
`;

exports[`App should match snapshot when isLoading is set to false 1`] = `
<div
  className="main"
>
  <div
    className="main-display"
  >
    <h1
      className="title"
    >
      Hello 
    </h1>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <DevicesContainer />
  </div>
</div>
`;
